src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php

```diff
@@ -113,13 +110,16 @@ protected function loadBlocks(array $config, ContainerBuilder $container)
         foreach ($blocks as $blockName => $block) {
             $container
                 ->register(
-                    'elcodi.sitemap_entity_loader.' . $blockName,
-                    '%elcodi.core.sitemap.loader.entity_loader.class%'
+                    'elcodi.sitemap_element_generator.entity_' . $blockName,
+                    'Elcodi\Component\Sitemap\Element\EntitySitemapElementGenerator'
                 )
+                ->addArgument(new Reference('elcodi.factory.sitemap_element'))
```

Comment:
Instead of many calls to `addArgument`, make it just one to `setArguments`.
Or even better (IMO) to create a `Definition` and call `setDefinition` directly.